api/lib/infrastructure/repositories/complementary-certification-habilitation-repository.js

```diff
@@ -1,15 +1,33 @@
 const { knex } = require('../../../db/knex-database-connection');
 
+const COMPLEMENTARY_CERTIFICATION_HABILITATIONS_TABLE_NAME = 'complementary-certification-habilitations';
+
 module.exports = {
   async save(complementaryCertification) {
     const columnsToSave = {
       complementaryCertificationId: complementaryCertification.complementaryCertificationId,
       certificationCenterId: complementaryCertification.certificationCenterId,
     };
-    return knex('complementary-certification-habilitations').insert(columnsToSave);
+    return knex(COMPLEMENTARY_CERTIFICATION_HABILITATIONS_TABLE_NAME).insert(columnsToSave);
   },
 
   async deleteByCertificationCenterId(certificationCenterId) {
-    return knex('complementary-certification-habilitations').delete().where({ certificationCenterId });
+    return knex(COMPLEMENTARY_CERTIFICATION_HABILITATIONS_TABLE_NAME).delete().where({ certificationCenterId });
+  },
+
+  async findByCertificationCenterId(certificationCenterId) {
+    return knex(COMPLEMENTARY_CERTIFICATION_HABILITATIONS_TABLE_NAME)
+      .select([
+        'complementary-certifications.id',
+        'complementary-certifications.key',
+        'complementary-certifications.label',
+      ])
+      .join(
+        'complementary-certifications',
+        'complementary-certifications.id',
+        `${COMPLEMENTARY_CERTIFICATION_HABILITATIONS_TABLE_NAME}.complementaryCertificationId`
+      )
+      .where(`${COMPLEMENTARY_CERTIFICATION_HABILITATIONS_TABLE_NAME}.certificationCenterId`, certificationCenterId)
+      .returning('*');
```

Comment:
Supprimer ce `returning` qui est en contradiction avec le `select` ci-dessus ?

```suggestion
      .where(`${COMPLEMENTARY_CERTIFICATION_HABILITATIONS_TABLE_NAME}.certificationCenterId`, certificationCenterId);
```